airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/factory.py

```diff
@@ -243,7 +243,7 @@ def is_object_definition_with_class_name(definition):
 
     @staticmethod
     def is_object_definition_with_type(definition):
-        return isinstance(definition, dict) and "type" in definition
+        return isinstance(definition, dict) and "type" in definition and definition["type"] != "object"
```

Comment:
this makes me slightly nervous, but the `connection_specification` is effectively just a mapping/dict/object so it gets read in as an object when we are generating the spec component. Because of that, we attempt to resolve it in the `CLASS_TYPES_REGISTRY` which will rightfully fail.

I'm not sure why we never saw this before, but I spot checked some components and didn't see any fields that were of mapping type so that might be why.